src/Uno.UI/UI/Xaml/Application.macOS.cs

```diff
@@ -110,31 +114,44 @@ private bool TryHandleUrlActivation(NSUrl[] urls, ApplicationExecutionState prev
 		/// <returns>System theme</returns>
 		private ApplicationTheme GetDefaultSystemTheme()
 		{
-			const string AutoSwitchKey = "AppleInterfaceStyleSwitchesAutomatically";
-			var autoChange = NSUserDefaults.StandardUserDefaults[AutoSwitchKey];
-			if (autoChange != null)
+			var version = GetSystemVersion();
+			if (version >= new Version(10, 14))
 			{
-				var autoChangeEnabled = NSUserDefaults.StandardUserDefaults.BoolForKey(AutoSwitchKey);
-				if (autoChangeEnabled)
+				var app = NSAppearance.CurrentAppearance?.FindBestMatch(new string[]
 				{
-					if (NSUserDefaults.StandardUserDefaults["AppleInterfaceStyle"] == null)
-					{
-						return ApplicationTheme.Dark;
-					}
-					else
-					{
-						return ApplicationTheme.Light;
-					}
+					NSAppearance.NameAqua,
+					NSAppearance.NameDarkAqua
+				});
+
+				if (app == NSAppearance.NameDarkAqua)
+				{
+					return ApplicationTheme.Dark;
 				}
 			}
-			if (NSUserDefaults.StandardUserDefaults["AppleInterfaceStyle"] == null)
-			{
-				return ApplicationTheme.Light;
-			}
-			else
+			return ApplicationTheme.Light;
+		}
+
+		private Version GetSystemVersion()
```

Comment:
That should probably go into another helper class in Uno, or an existing API that provides that.

Reply:
Will check if it is somewhere 👍

Reply:
Fixed